src/utils/Render.js

```diff
@@ -115,23 +135,110 @@ function renderWithStrategy(strategy) {
     // cache the element
     const Element = React.createElement(Container, props)
 
-    // render so we kick things off and get the props
-    buffer.fill(Element)
+    const start = Date.now()
+
+    return buffer.render(alt, Element, info).then((obj) => {
+      const time = Date.now() - start
 
-    // flush out the results in the buffer synchronously setting the store
-    // state and returning the markup
-    return buffer.flush(alt, Element)
+      return {
+        error: obj.error,
+        html: obj.html,
+        state: obj.state,
+        fulfilled: obj.fulfilled,
+        element: obj.element,
+        diagnostics: {
+          iterations: obj.diagnostics.iterations,
+          dispatches: obj.diagnostics.dispatches,
+          time
+        }
+      }
+    })
   }
 }
 
-export function toDOM(Component, props, documentNode, shouldLock) {
-  const buffer = new DispatchBuffer()
-  buffer.locked = !!shouldLock
-  const Node = usingDispatchBuffer(buffer, Component)
-  const Element = React.createElement(Node, props)
-  buffer.clear()
-  return React.render(Element, documentNode)
-}
+export default class Render {
+  constructor(alt, options = {}) {
+    this.alt = alt
+    this.options = options
+
+    // defaults
+    // 500ms or 5 iteration max
+    this.options.timeout = options.timeout || 500
+    this.options.maxIterations = options.maxIterations || 5
+  }
 
-export const toStaticMarkup = renderWithStrategy('renderToStaticMarkup')
-export const toString = renderWithStrategy('renderToString')
+  toString(Component, props) {
+    this.options.i = 0
+    this.options.time = 0
+    return renderWithStrategy('renderToString')(
+      this.alt,
+      Component,
+      props,
+      this.options
+    )
+  }
+
+  toStaticMarkup(Component, props) {
+    this.options.i = 0
+    this.options.time = 0
+    return renderWithStrategy('renderToStaticMarkup')(
+      this.alt,
+      Component,
+      props,
+      this.options
+    )
+  }
+
+  toDOM(Component, props, documentNode, opts = {}) {
+    const buffer = new DispatchBuffer()
+
+    if (opts.fulfilled) buffer.fulfilled = opts.fulfilled
+    const Node = usingDispatchBuffer(buffer, Component)
+    const Element = React.createElement(Node, props)
+    buffer.clear()
+    return React.render(Element, documentNode)
+  }
+
+  static resolve(fetch, MaybeComponent) {
+    function bind(Component) {
+      return React.createClass({
```

Comment:
Not a bad idea...

Although we do have https://github.com/goatslacker/alt/blob/master/src/utils/statics.js for that sort of thing.

I'll amend the changes.